packages/story-editor/src/app/story/effects/useHistoryEntry.js

```diff
@@ -23,28 +23,83 @@ import { useEffect, useRef } from '@web-stories-wp/react';
  * Internal dependencies
  */
 import { useHistory } from '../../history';
+import deleteNestedKeys from '../utils/deleteNestedKeys';
+
+// @todo REMOVE ROTATION ANGLE! For testing only!
+const ELEMENT_PROPS_TO_IGNORE = ['resource.baseColor', 'rotationAngle'];
 
 // Record any change to core variables in history (history will know if it's a replay)
 function useHistoryEntry({ story, current, pages, selection, capabilities }) {
   const {
+    state: { currentEntry },
     actions: { stateToHistory },
   } = useHistory();
 
+  const currentHistoryEntryRef = useRef();
+  useEffect(() => {
+    if (currentEntry) {
+      currentHistoryEntryRef.current = JSON.parse(JSON.stringify(currentEntry));
+    }
+  }, [currentEntry]);
+
   const currentPageIndexRef = useRef();
   const selectedElementIdsRef = useRef();
   useEffect(() => {
     currentPageIndexRef.current = current;
     selectedElementIdsRef.current = selection;
   }, [current, selection]);
 
-  useEffect(() => {
-    stateToHistory({
-      story,
-      current: currentPageIndexRef.current,
-      selection: selectedElementIdsRef.current,
-      pages,
-      capabilities,
+  const deleteKeysFromPages = (list) => {
+    // Create a copy of the list not to influence the original.
+    return JSON.parse(JSON.stringify(list)).map((page) => {
```

Comment:
Have you checked how expensive this is? Maybe a regular `deepClone()` function is better because I fear this is quite computationally expensive.

Reply:
No, I haven't yet, I'll try it out.

I was also wondering about the speed here but it needs to be done somehow anyway -- within reducer or not. I'll do some testing.

Reply:
Tested it out using `clone-deep` package vs stringifying using a 40 page story with various elements including a ridiculous amount of media on the pages, and it looks like even in these extreme conditions, the cloning takes approximately 2ms for both. Sometimes, however, when adding a new element, the stringifying spiked to 5ms.

When using a normal story (20 pages, only a few media elements per page), it the tracker consistently got 1ms for this. Considering that it will skip creating a new history entry, that'll probably be an improvement.

I'll go with using `clone-deep` since `JSON.stringify` can potentially break if we should start storing other type of data in the objects (other than strings and numbers, that is).